src/NServiceBus.Core/Routing/RoutingPluginSettingsExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace NServiceBus
+{
+    using Features;
+    using Routing;
+
+    /// <summary>
+    /// Provides convinient API for plugins that need to extend the routing.
+    /// </summary>
+    public static class RoutingPluginSettingsExtensions
+    {
+        /// <summary>
+        /// Returns the routing table.
+        /// </summary>
+        /// <param name="featureConfigurationContext">Context.</param>
+        public static UnicastRoutingTable RoutingTable(this FeatureConfigurationContext featureConfigurationContext)
+        {
+            return featureConfigurationContext.Settings.Get<UnicastRoutingTable>();
+        }
+
+        /// <summary>
+        /// Returns the routing table.
+        /// </summary>
+        /// <param name="featureConfigurationContext">Context.</param>
+        public static EndpointInstances EndpointInstances(this FeatureConfigurationContext featureConfigurationContext)
+        {
+            return featureConfigurationContext.Settings.Get<EndpointInstances>();
+        }
+
+        /// <summary>
+        /// Returns the distribution policy.
+        /// </summary>
+        /// <param name="featureConfigurationContext">Context.</param>
+        public static DistributionPolicy DistributionPolicy(this FeatureConfigurationContext featureConfigurationContext)
+        {
+            return featureConfigurationContext.Settings.Get<DistributionPolicy>();
+        }
+    }
+}
```

Comment:
still not sure how much sense it makes to have these extensions on the feature context. It can still confuse users on non-msmq transports and it's not very discoverable ("I want to do custom routing, that must obviously be possible in a feature setup..."?). Then we can make that API also somehow available on the endpoint config?